src/content/en/fundamentals/primers/modules.md

```diff
@@ -0,0 +1,578 @@
+`import()`](/web/updates/2017/11/dynamic-import) makes this possible!
+
+```html
+<script type="module">
+  (async () => {
+    const moduleSpecifier = './lib.mjs';
+    const {repeat, shout} = await import(moduleSpecifier);
+    repeat('hello');
+    // → 'hello hello'
+    shout('Dynamic import in action');
+    // → 'DYNAMIC IMPORT IN ACTION!'
+  })();
+</script>
+```
+
+Unlike static `import`, dynamic `import()` can be used from within regular scripts. It’s an easy
+way to incrementally start using modules in your existing code base. For more details, see [our
+article on dynamic `import()`](/web/updates/2017/11/dynamic-import).
+
+Note: [webpack has its own version of
+`import()`](/web/fundamentals/performance/webpack/use-long-term-caching) that cleverly splits the
+imported module into its own chunk, separate from the main bundle.
+
+### `import.meta` {: #import-meta }
+
+Another new module-related feature is `import.meta`, which gives you metadata about the current
+module. The exact metadata you get is not specified as part of ECMAScript; it depends on the host
+environment. In a browser, you might get different metadata than in Node.js, for example.
+
+Here’s an example of `import.meta` on the web. By default, images are loaded relative to the
+current URL in HTML documents. `import.meta.url` makes it possible to load an image relative to
+the current module instead.
+
+```js
+function loadThumbnail(relativePath) {
+  const url = new URL(relativePath, import.meta.url);
+  const image = new Image();
+  image.src = url;
+  return image;
+}
+
+const thumbnail = loadThumbnail('../img/thumbnail.png');
+container.append(thumbnail);
+```
+
+## Performance recommendations  {: #performance }
+
+### Keep bundling {: #bundle }
+
+With modules, it becomes possible to develop websites without using bundlers such as webpack,
+Rollup, or Parcel. It’s fine to use native JS modules directly in the following scenarios:
+
+- during local development
+- in production for small web apps with less than 100 modules in total and with a relatively
+    shallow dependency tree (i.e. a maximum depth less than 5)
+
+However, as we learned during [our bottleneck analysis of Chrome’s loading pipeline when loading a
+modularized library composed of ~300
+modules](https://docs.google.com/document/d/1ovo4PurT_1K4WFwN2MYmmgbLcr7v6DRQN67ESVA-wq0/pub), the
+loading performance of bundled applications is better than unbundled ones.
+
+[![](imgs/renderer-main-thread-time-breakdown.png)](https://docs.google.com/document/d/1ovo4PurT_1K4WFwN2MYmmgbLcr7v6DRQN67ESVA-wq0/pub)
+
+One reason for this is that the static `import`/`export` syntax is statically analyzable, and it
+can thus help bundler tools optimize your code by eliminating unused exports. Static `import` and
+`export` are more than just syntax; they are a critical tooling feature!
+
+**Our general recommendation is to continue using bundlers before deploying modules to
```

Comment:
Total nit - you could use `Success: We recommend you use bundlers...` Success is a special callout similar to note that marks the whole block green with a checkmark to the right

Reply:
Good call! Done.